src/Wallet/Cip30Mock.purs

```diff
@@ -0,0 +1,134 @@
+module Wallet.Cip30Mock where
+
+import Prelude
+
+import Contract.Monad (Contract, ContractEnv, wrapContract)
+import Control.Monad.Error.Class (liftMaybe, try)
+import Control.Monad.Reader (ask)
+import Control.Monad.Reader.Class (local)
+import Control.Promise (Promise, fromAff)
+import Data.Either (hush)
+import Data.Foldable (fold)
+import Data.Lens ((.~))
+import Data.Lens.Common (simple)
+import Data.Lens.Iso.Newtype (_Newtype)
+import Data.Lens.Record (prop)
+import Data.Map as Map
+import Data.Maybe (Maybe(Just))
+import Data.Newtype (unwrap)
+import Deserialization.Transaction (deserializeTransaction)
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Effect.Aff.Class (liftAff)
+import Effect.Class (liftEffect)
+import Effect.Exception (error)
+import Effect.Unsafe (unsafePerformEffect)
+import Helpers (liftEither)
+import QueryM (QueryM, runQueryMInRuntime)
+import QueryM.Utxos (utxosAt)
+import Serialization (convertTransactionUnspentOutput, convertValue, toBytes)
+import Serialization.WitnessSet (convertWitnessSet)
+import Type.Proxy (Proxy(Proxy))
+import Types.ByteArray (byteArrayToHex, hexToByteArray)
+import Types.CborBytes (cborBytesFromByteArray)
+import Untagged.Union (asOneOf)
+import Wallet (Wallet, mkFlintWalletAff, mkGeroWalletAff, mkNamiWalletAff)
+import Wallet.Key
+  ( KeyWallet(KeyWallet)
+  , PrivatePaymentKey
+  , PrivateStakeKey
+  , privateKeysToKeyWallet
+  )
+
+data WalletMock = MockFlint | MockGero | MockNami
+
+-- | Construct a CIP-30 wallet mock that exposes `KeyWallet` functionality
```

Comment:
I think it would be nice to include some documentation on how to use this (the complete process)

Reply:
Currently if someone searches the repository for the mock wallet they'll probably just find this comment